### PR TITLE
ci: added unit test infrastructure

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

**Review by trewest — DISMISSED** [review]
/lgtm

File: .testcoverage.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#
+# For cases where there are many coverage profiles, such as when running
+# unit tests and integration tests separately, you can combine all those
+# profiles into one. In this case, the profile should have a comma-separated list
+# of profile files, e.g., 'cover_unit.out,cover_integration.out'.
+profile: cover.out
+
+# (optional; but recommended to set)
+# When specified reported file paths will not contain local prefix in the output
+local-prefix: "github.com/kononovn/lsp"
```
Comment (sebrandon1) [inline .testcoverage.yml:12]: Is this a leftover from testing?

**Review by sebrandon1 — COMMENTED** [review]
Looks good, one comment.

File: .testcoverage.yml (hunk 1 above)
Comment (cdvultur) [reply .testcoverage.yml:12]: Indeed, this seems the only issue, otherwise lgtm

**Pull request merged**